src/charm.py

```diff
@@ -62,30 +59,33 @@ def _on_config_changed(self, event):
             logger.info("Invalid port number, 22 is reserved for SSH")
             return
 
-        self._update_layer_and_restart(None)
+        self._update_layer_and_restart()
+
+    def _on_demo_server_pebble_ready(self, event: ops.PebbleReadyEvent) -> None:
+        self._update_layer_and_restart()
 
     def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
         """Event is fired when postgres database is created."""
-        self._update_layer_and_restart(None)
+        self._update_layer_and_restart()
 
-    def _update_layer_and_restart(self, event) -> None:
+    def _update_layer_and_restart(self) -> None:
         """Define and start a workload using the Pebble API.
 
         You'll need to specify the right entrypoint and environment
         configuration for your specific workload. Tip: you can see the
         standard entrypoint of an existing container using docker inspect
 
+        Learn more about interacting with Pebble at https://juju.is/docs/sdk/pebble
```

Comment:
Same comment as chapter 1: seems like the Pebble docs site would be better than the README on the line below this.